csm_web/scheduler/email/email_utils.py

```diff
@@ -0,0 +1,217 @@
+import os
+import base64
+from email.message import EmailMessage
+
+from django.conf import settings
+
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+from .email_errors import EmailFormattingError, NoEmailError, EmailAuthError
+
+
+def email_enroll(student):
+    """Sends the corresponding mentor an email notification that the STUDENT enrolled in their section"""
+
+    try:
+        mentor = student.section.mentor
+        course_title = student.course.title
+        mentor_name = mentor.user.first_name + " " + mentor.user.last_name
+        student_name = student.user.first_name + " " + student.user.last_name
+        mentor_email = mentor.user.email
+        student_email = student.user.email
+    except AttributeError:
+        raise NoEmailError
+
+    if mentor.user.subscribed:
+        body = _render_template("student_enroll/notif.html").format(
+            course_title=course_title,
+            mentor_name=mentor_name,
+            student_name=student_name,
+        )
+        _email_send_message(f"[CSM - {course_title}] Student Enrolled", body, mentor_email)
+
+    if student.user.subscribed:
+        body = _render_template("student_enroll/confirm.html").format(
+            course_title=course_title,
+            mentor_name=mentor_name,
+            student_name=student_name,
+        )
+        _email_send_message(f"[CSM - {course_title}] Section Enrollment Confirmation", body, student_email)
```

Comment:
There are potential HTML injection issues here, since you're just directly inserting user-provided text into HTML. If somebody just created a user with a malicious payload as the name, this could result in very bad security vulnerabilities.

I'm unsure whether there are existing libraries that will do the text escaping, or if the Gmail API has some functions for email templates, but this should be looked into.